README.md

```diff
@@ -115,225 +98,18 @@ And destructuring tuples with `let`:
-Nil  =        @#List c @#List n n
-```
-
-When encoding the pattern matching, the application can then use the same label:
-
-```rs
-// not = @bool match bool { T: (F) F: (T) } 
-not = @bool (#Bool bool F T)
-```
-
-This allows, in some limited* scenarios, automatic vectorization:
-```rs
-// vectorizes to: (Cons F (Cons T (Cons F Nil)))
-main = (not (Cons T (Cons F (Cons T Nil))))
-```
-
-The tagged lambda and applications are compiled to `dup` `inet` nodes with different tag values for each label. This allows for different commutation rules, check [HVM-Core](https://github.com/HigherOrderCO/hvm-core/tree/main#language) to learn more about it.
-
-### *limitations:
-- When using a built-in ADT (created using the `data` keyword), it must not contain fields 
-- The function must not be recursive
-- There must not be labels in common between the function and what you want to vectorize over
-
-## Compilation of Terms to HVM-core
-
-How terms are compiled into interaction net nodes?
-
-HVM-Core has a bunch of useful nodes to write IC programs.
-Every node contains one `main` port `0` and two `auxiliary` ports, `1` and `2`.
-
-There are 6 kinds of nodes, Erased, Constructor, Reference, Number, Operation and Match.
-
-A lambda `λx x` compiles into a Constructor node.
-An application `((λx x) (λx x))` also compiles into a Constructor node.
-We differentiate then by using the ports.
-
-```
-  0 - Points to the lambda occurrence      0 - Points to the function
-  |                                        |
-  λ                                        @
- / \                                      / \
-1   2 - Points to the lambda body        1   2 - Points to the application occurrence
-|                                        |
-Points to the lambda variable            Points to the argument
-```
-
-So, if we visit a Constructor at port 0 it's a Lambda, if we visit at port 2 it's an Application.
-
-Also, nodes have labels, we use the label to store data in the node's memory and also differentiate them.
-
-The `Number` node uses the label to store it's number.
-An `Op2` node uses the label to store it's operation.
-And a `Constructor` node can have a label too! The label is used for `Dup` and `Tuple` nodes.
+### Advanced features
 
-Check [HVM-Core](https://github.com/HigherOrderCO/hvm-core/tree/main#language) to know more about.
+Advanced features are described in other files:
 
-### Runtime and Compiler optimizations
+- Duplications and superpositions: [Dups and sups](docs/dups-and-sups.md)
+- Data types: [Defining data types](docs/defining-data-types.md)
+- Native numbers and operations: [Native numbers](docs/native-numbers.md)
+- Scopeless lambdas: [Using scopeless lambdas](docs/using-scopeless-lambdas.md)
+- Tagged lambdas and applications: [Automatic vectorization with tagged lambdas](docs/automatic-vectorization-with-tagged-lambdas.md)
+- Lazy definitions: [Making recursive definitions lazy](docs/lazy-definitions.md)
```

Comment:
We should probably order this by complexity/"advancedness".

Data types, numbers and lazy definitions should come first, since they are important for writing even normal lambda programs.